web-api/src/ingestion/routes.rs

```diff
@@ -326,21 +331,29 @@ async fn upsert_documents(
         .partition_map::<Vec<_>, Vec<_>, _, _, _>(|document| {
             let (data, is_candidate) = existing_documents
                 .get(&document.id)
-                .map(|(snippet, properties, tags, is_candidate)| {
-                    ((snippet, properties, tags), *is_candidate)
-                })
+                .map(
+                    |(snippet, was_summarized, properties, tags, is_candidate)| {
+                        ((snippet, was_summarized, properties, tags), *is_candidate)
+                    },
+                )
                 .unzip();
 
-            let new_snippet = data.map_or(true, |(snippet, _, _)| snippet != &document.snippet);
+            let new_snippet = data.map_or(true, |(snippet, was_summarized, _, _)| {
```

Comment:
:wrench:  as discussed, this should have been untouched. Having to add a maintain code, even if it is a single line, to avoid recomputing the snippet for a few tens of documents is not worth it.